dimagi/utils/django/management.py

```diff
@@ -11,3 +16,40 @@ def are_you_sure(prompt="Are you sure you want to proceed? (yes or no): "):
     except Exception:
         return False
 
+
+def export_as_csv_action(description="Export selected objects as CSV file",
+                         fields=None, exclude=None, header=True):
+    """
+    Include this in ModelAdmin.actions to get an option of export as CSV in Django Admin.
+
+    copied from https://djangosnippets.org/snippets/2369/
+
+    This function returns an export csv action
+    'fields' and 'exclude' work like in django ModelForm
+    'header' is whether or not to output the column names as the first row
+    """
+    def export_as_csv(modeladmin, request, queryset):
+        """
+        Generic csv export admin action.
+        based on http://djangosnippets.org/snippets/1697/
+        """
+        opts = modeladmin.model._meta
+        field_names = set([field.name for field in opts.fields])
```

Comment:
Minor point, but you don't need to make a list here, this can just be
```python
field_names = set(field.name for field in opts.fields)
# or
field_names = {field.name for field in opts.fields}
```
(no action needed)